persistence-api/src/main/scala/com/evolutiongaming/akkaeffect/persistence/EventStore.scala

```diff
@@ -7,14 +7,16 @@ trait EventStore[F[_], A] extends EventStore.Read[F, A] with EventStore.Write[F,
 object EventStore {
 
   trait Read[F[_], A] {
-    def from(seqNr: SeqNr): F[sstream.Stream[F, Event[A]]]
+    def events(fromSeqNr: SeqNr): F[sstream.Stream[F, Persisted[A]]]
   }
 
   trait Write[F[_], A] {
-    def save(events: Events[A]): F[F[SeqNr]]
+    def save(events: Events[Event[A]]): F[F[SeqNr]]
     def deleteTo(seqNr: SeqNr): F[F[Unit]]
   }
 
-  final case class Event[A](event: A, seqNr: SeqNr)
+  sealed trait Persisted[+A]
```

Comment:
I like your way better, because it simplifies the most common use case, but is not ideal, still.

The problem is that the following API implies `HighestSeqNr` may come anywhere between `Event` instances, while, in reality, at least for Akka, it may only come last, i.e. when `JournalProtocol.RecoverySuccess` comes:
```scala
def events(fromSeqNr: SeqNr): F[Stream[F, Persisted[A]]]
```

Reply:
True, the API does not limit ordering there. Do you think that the ordering is important? AFAIK, in `EventSourcedActorOf`, `HighestSeqNr` will be correctly handled even if it comes in the middle of the stream (under condition that the `seqNr` always increasing). Might be reasonable to rename it to `SeqNrEvent` or similar thus removing notion of "highest"?

Reply:
I slept over it and I think we can handle / discuss it separately, eventually, to not block the current effort. 

I do not have the better idea at hand, right now.